KomodoPlatform/atomicDEX-Desktop
# Fix segwit bestorders

Closes https://github.com/KomodoPlatform/atomicDEX-Desktop/issues/2236
Fixes issue reported by @cipig in https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/2230#discussion_r1155283698

To test:
- Login, go to dex pro page and select an order from the orderbook for a popular pair. 
- Wait for bestorders table to populate, then scroll through it
- You should see both the legacy and segwit version of a coin where supported (LTC or DGB should be there)
- Wait a minute or so without clicking anything to give the bestorders list time to refresh.
- Confirm both the legacy and segwit version are still present.
- Click on the segwit option of one of the coins in the bestorders list
- Confirm it changes the pair and loads a fresh orderbook as expected.
- Confirm doing a trade after this correctly sends or receives funds from the segwit address.

![image](https://github.com/KomodoPlatform/atomicDEX-Desktop/assets/35845239/fae114c5-d648-4027-bedb-3946c50db156)
![image](https://github.com/KomodoPlatform/atomicDEX-Desktop/assets/35845239/6d466b6a-33b0-4e65-8a3b-03da4705f785)

**Note**: Some crashes in the past were a result of an assert statement or different number of items in `m_model_data` and `m_orders_id_registry` - this includes where the legacy and segwit orders which had the same UUID (shared orderbook since https://github.com/KomodoPlatform/atomicDEX-API/issues/1598) and would omit one because uuid was an object key. Adding the `-segwit` suffix to the uuid prevents this. I have checked a lot of code to ensure this suffix has no unintended consequences.

This PR also sets the following coins to be activated by default to let users know about the segwit functionality:
- BTC-segwit
- LTC-segwit
- DGB-segwit

It might be an opportunity for some cross-promotion with he DGB team @tonymorony
